cmd/drand-cli/daemon.go

```diff
@@ -12,44 +12,59 @@ import (
 
 func startCmd(c *cli.Context) error {
 	conf := contextToConfig(c)
+
+	// Create and start drand daemon
+	drandDaemon, err := core.NewDrandDaemon(conf)
+	if err != nil {
+		return fmt.Errorf("can't instantiate drand daemon %s", err)
+	}
+
+	// Load possible existing stores
 	stores, err := key.NewFileStores(conf.ConfigFolderMB())
 	if err != nil {
-		return fmt.Errorf("can't read file stores %s", err)
+		return err
 	}
 
-	var drand *core.Drand
+	for beaconID, fs := range stores {
+		var bp *core.BeaconProcess
 
-	// determine if we already ran a DKG or not
-	beaconID, fs := key.GetFirstStore(stores)
-	_, errG := fs.LoadGroup()
-	_, errS := fs.LoadShare()
+		_, errG := fs.LoadGroup()
+		_, errS := fs.LoadShare()
 
-	// XXX place that logic inside core/ directly with only one method
-	freshRun := errG != nil || errS != nil
+		// XXX place that logic inside core/ directly with only one method
+		freshRun := errG != nil || errS != nil
```

Comment:
Can we move that logic inside beacon processes ? They should be responsible for deciding if they are launched or not, not the daemon.
Or at the very least create an issue about it so we can address it after this one - I've put that comment already in the first PR IIRC.

Reply:
This was partially fixed on a previous commit. The issue #860 address this.